config/secure-headers.php

```diff
@@ -316,47 +316,227 @@
+		'frame-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src
+		'img-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/manifest-src
+		'manifest-src' => [
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/media-src
+		'media-src' => [
+			'self' => true,
+			'allow' => [
+				'blob:',
+			],
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/navigate-to
+		'navigate-to' => [
+			'unsafe-allow-redirects' => false,
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/object-src
+		'object-src' => [
+			'none' => true,
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/plugin-types
+		'plugin-types' => [
+			// 'application/pdf',
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/prefetch-src
+		'prefetch-src' => [
+		],
+
+		// https://w3c.github.io/webappsec-trusted-types/dist/spec/#integration-with-content-security-policy
+		'require-trusted-types-for' => [
+			'script' => false,
+		],
+
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/sandbox
+		'sandbox' => [
+			'enable' => false,
+
+			'allow-downloads-without-user-activation' => false,
+
+			'allow-forms' => false,
+
+			'allow-modals' => false,
+
+			'allow-orientation-lock' => false,
+
+			'allow-pointer-lock' => false,
+
+			'allow-popups' => false,
+
+			'allow-popups-to-escape-sandbox' => false,
+
+			'allow-presentation' => false,
+
+			'allow-same-origin' => false,
+
+			'allow-scripts' => false,
+
+			'allow-storage-access-by-user-activation' => false,
+
+			'allow-top-navigation' => false,
+
+			'allow-top-navigation-by-user-activation' => false,
+		],
 
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src
 		'script-src' => [
+			'none' => false,
+
+			'self' => true,
+
+			// https://www.chromestatus.com/feature/5792234276388864
+			'report-sample' => true,
+
 			'allow' => [
 				'https://www.dropbox.com/static/api/1/dropins.js',
 				// 'url',
 			],
+
+			'schemes' => [
+				// 'data:',
+				// 'https:',
+			],
+
+			/* followings are only work for `script` and `style` related directives */
+
+			'unsafe-inline' => false,
+
+			'unsafe-eval' => false,
+
+			// https://www.w3.org/TR/CSP3/#unsafe-hashes-usage
+			'unsafe-hashes' => false,
+
+			// Enable `strict-dynamic` will *ignore* `self`, `unsafe-inline`,
+			// `allow` and `schemes`. You can find more information from:
+			// https://www.w3.org/TR/CSP3/#strict-dynamic-usage
+			'strict-dynamic' => false,
+
 			'hashes' => [
 				'sha256' => [
+					// 'sha256-hash-value-with-base64-encode',
 					'8bLztrDF3NUpheSuvAzpebgX1DpPJEfhmUHKTwGF4qA=',
```

Comment:
Where does this hash come from? It results into the following header

```
script-src 'self' 'report-sample' 'sha256-8bLztrDF3NUpheSuvAzpebgX1DpPJEfhmUHKTwGF4qA=' https://www.dropbox.com/static/api/1/dropins.js;
```

According to the docs, this is an alternative way to allow a script without denoting its source by giving its hash, i.e. the script with this hash value is allowed. Do we need it? If yes, we should add a comment whose hash value is this and how this hash is re-computed whenever the script to which is hash belongs is updated.

Otherwise we should remove it.

Reply:
It was added in https://github.com/LycheeOrg/Lychee/commit/3e6cda2cb725b5f0affbfbbbc68cbe22e7dcb1b4. Not sure what for though.

Reply:
I tried to remove it, didn't seem to break anything. So I guess it will stay removed.

Reply:
Removed in [3f5583f](https://github.com/LycheeOrg/Lychee/pull/1528/commits/3f5583f3f4300551dbb90229ed1d7e908202272d)